$NetBSD: patch-src_Makefile.OCaml,v 1.1 2019/07/29 18:07:10 wiz Exp $

Correct lablgtk paths for 2.16, and use ocamlopt.opt for compilation
Add -unsafe-string for OCaml >= 4.06.

--- src/Makefile.OCaml.orig	2015-10-05 18:14:23.000000000 +0000
+++ src/Makefile.OCaml
@@ -75,8 +75,8 @@ OCAMLLIBDIR=$(shell ocamlc -v | tail -1 
 #
 # This should be set to an appropriate value automatically, depending
 # on whether the lablgtk library is available
-LABLGTKLIB=$(OCAMLLIBDIR)/lablgtk
-LABLGTK2LIB=$(OCAMLLIBDIR)/lablgtk2
+LABLGTKLIB=$(OCAMLLIBDIR)/site-lib/lablgtk
+LABLGTK2LIB=$(OCAMLLIBDIR)/site-lib/lablgtk2
 ##BCP [3/2007]: Removed temporarily, since the OSX UI is not working well 
 ## at the moment and we don't want to confuse people by building it by default
 ifeq ($(OSARCH),osx)
@@ -292,14 +292,14 @@ endif
 
 # Gtk GUI
 ifeq ($(UISTYLE), gtk)
-  CAMLFLAGS+=-I +lablgtk
+  CAMLFLAGS+=-I $(OCAMLLIBDIR)/site-lib/lablgtk
   OCAMLOBJS+=pixmaps.cmo uigtk.cmo linkgtk.cmo
   OCAMLLIBS+=lablgtk.cma
 endif
 
 # Gtk2 GUI
 ifeq ($(UISTYLE), gtk2)
-  CAMLFLAGS+=-I +lablgtk2
+  CAMLFLAGS+=-I $(OCAMLLIBDIR)/site-lib/lablgtk2
   OCAMLOBJS+=pixmaps.cmo uigtk2.cmo linkgtk2.cmo
   OCAMLLIBS+=lablgtk.cma
 endif
@@ -384,7 +384,7 @@ ifeq ($(PROFILING), true)
 else
   OCAMLC=$(EXEC_PREFIX)ocamlc
 endif
-OCAMLOPT=$(EXEC_PREFIX)ocamlopt
+OCAMLOPT=$(EXEC_PREFIX)ocamlopt.opt
 
 ifeq ($(NATIVE), true)
   ## Set up for native code compilation
